hdate/tekufot.py

```diff
@@ -0,0 +1,346 @@
+            ("shemini_to_cheshvan", self.shemini_atzeret, self.seventh_cheshvan),
+            ("cheshvan_to_geshamim", self.seventh_cheshvan, self.cheilat_geshamim),
+            ("geshamim_to_pessah", self.cheilat_geshamim, self.pesach),
+            ("pessah_to_shemini_next", self.pesach, self.next_shemini_atzeret),
+        ]
+
+    def get_prayer_for_date(
+        self,
+        date: Union[datetime.date, str, datetime.datetime] = datetime.datetime.now(),
+        tradition: str = "israel",
+        language: str = "english",
+    ) -> str:
+        """
+        Returns the appropriate prayer phrases for the given date, tradition,
+        and language. The tradition can be
+        'israel',
+        'diaspora_ashkenazi',
+        or 'diaspora_sephardi'.
+        The language can be 'english', 'french', or 'hebrew'.
+        """
+        if isinstance(date, str):
+            # If a string is given, parse it into a date/datetime if needed.
+            # For now, assume YYYY-MM-DD format.
+            date = datetime.datetime.strptime(date, "%Y-%m-%d").date()
+        elif isinstance(date, datetime.datetime):
+            date = date.date()
+
+        if date is None:
+            date = self.date
+
+        for period_name, start_date, end_date in self.prayer_periods:
+            if start_date and end_date:
+                if start_date <= date < end_date:
+                    tradition_data = PRAYER_DESCRIPTIONS.get(period_name)
+                    if tradition_data:
+                        lang_data = getattr(tradition_data, tradition, None)
+                        if lang_data:
+                            return getattr(
+                                lang_data, language, "No phrase for this language."
+                            )
+        return "No prayer phrase found for this date, tradition, and language."
+
+    def get_results(self) -> dict[str, object]:
+        """
+        Returns a dictionary containing all the calculated dates and times.
+        """
+        return {
+            "date": self.date,
+            "gregorian_year": self.gregorian_year,
+            "hebrew_year": self.hebrew_year,
+            "hebrew_year_p": self.hebrew_year_p,
+            "diaspora": self.diaspora,
+            "tekufa_tishrei": getattr(self, "tekufa_tishrei", None),
+            "tekufa_tevet": getattr(self, "tekufa_tevet", None),
+            "tekufa_nissan": self.tekufa_nissan,
+            "tekufa_tammuz": getattr(self, "tekufa_tammuz", None),
+            "cheilat_geshamim": getattr(self, "cheilat_geshamim", None),
+            "seventh_cheshvan": getattr(self, "seventh_cheshvan", None),
+            "prev_pesach": getattr(self, "prev_pesach", None),
+            "shemini_atzeret": getattr(self, "shemini_atzeret", None),
+            "pesach": getattr(self, "pesach", None),
+            "next_shemini_atzeret": getattr(self, "next_shemini_atzeret", None),
+        }
```

Comment:
If all results are part of self, there's no need for this method.

Reply:
i will remove it after . it is for debugging